torch/_torch_docs.py

```diff
@@ -8174,6 +8174,45 @@ def merge_dicts(*dicts):
 """,
 )
 
+add_docstr(
+    torch.narrow_copy,
+    r"""
+narrow_copy(input, dim, start, length, *, out=None) -> Tensor
+
+Same as :meth:`Tensor.narrow` except returning a copy rather
+than shared storage.  This is primarily for sparse tensors, which
+do not have a shared-storage narrow method.  Calling ``narrow_copy``
+with ``dimemsion > self.sparse_dim()`` will return a copy with the
+relevant dense dimension narrowed, and ``self.shape`` updated accordingly.
```

Comment:
I know this text was brought over from the existing docs, but since we are updating it:


The last sentence here seems to imply that one can only apply `narrow_copy` to dense dimensions when the input is sparse. This is not true,  as sparse dims may also be narrowed this way.

Ex:

```python
>>> import torch
>>> s = torch.ones(2,2,2,2).to_sparse(2)
>>> s
tensor(indices=tensor([[0, 0, 1, 1],
                       [0, 1, 0, 1]]),
       values=tensor([[[1., 1.],
                       [1., 1.]],

                      [[1., 1.],
                       [1., 1.]],

                      [[1., 1.],
                       [1., 1.]],

                      [[1., 1.],
                       [1., 1.]]]),
       size=(2, 2, 2, 2), nnz=4, layout=torch.sparse_coo)
>>> s.narrow_copy(0, 0, 1)
tensor(indices=tensor([[0, 0],
                       [0, 1]]),
       values=tensor([[[1., 1.],
                       [1., 1.]],

                      [[1., 1.],
                       [1., 1.]]]),
       size=(1, 2, 2, 2), nnz=2, layout=torch.sparse_coo)
>>> s.narrow_copy(3, 0, 1)
tensor(indices=tensor([[0, 0],
                       [0, 1]]),
       values=tensor([[[1., 1.],
                       [1., 1.]],

                      [[1., 1.],
                       [1., 1.]]]),
       size=(1, 2, 2, 2), nnz=2, layout=torch.sparse_coo)

```

Reply:
I've removed the reference to dense and added a sparse example